src/extension/extension.ts

```diff
@@ -545,7 +545,7 @@ export async function activate(context: vs.ExtensionContext, isRestart: boolean
 	if (testDiscoverer)
 		context.subscriptions.push(testDiscoverer);
 	const vsCodeTestController = vs.tests?.createTestController !== undefined // Feature-detect for Theia
-		? new VsCodeTestController(logger, testModel, testDiscoverer)
+		? new VsCodeTestController(logger, testModel, testDiscoverer, workspaceContext.config.forceFlutterWorkspace)
```

Comment:
I'd consider adding another flag (that's just also set in the same place as this one) that's descriptive specifically for this purpose. That way if the capabilities every change internally (or we want to reuse this for other types of workspaces), it's a bit simpler.

We actually have another flag somewhere that is the opposite of this (which disables it for the Dart VM). Perhaps if we add a specific flag, I could change that to use this (eg. it's `true` here, `false` for the Dart SDK, and `undefined` when we should check for pubspec).

I'm not sure exactly how to describe what it's doing - "pub run test" is a bit misleading here (since it's actually "dart test" and "flutter test" now).. Perhaps `supportsPackageTest` would be reasonable (both `dart test` and `flutter test` use `package:test`, which is what is providing the ability to run specific tests).

Reply:
Thanks for the name suggestion! I'll leave it up to you if you want to change other spots to using this.